eng/ci/templates/official/jobs/build-core-tools-host.yml

```diff
@@ -0,0 +1,123 @@
+jobs:
+- job: BuildCoreToolsHostWindows
+  displayName: '[Windows] Build CoreToolsHost'
+  pool:
+    name: 1es-pool-azfunc
+    image: 1es-windows-2022
+    os: windows
+
+  steps:
+  - task: UseDotNet@2
+    inputs:
+      version: 9.x
+      includePreviewVersions: true
+    displayName: Install .NET 9
+  - task: UseDotNet@2
+    inputs:
+      version: 6.x
+    displayName: Install .NET 6
+
+  - task: DotnetCoreCLI@2
+    displayName: Dotnet Publish (win-x64)
+    inputs:
+      command: publish
+      publishWebProjects: false
+      zipAfterPublish: false
+      arguments: -c Release -r win-x64 -o $(Build.SourcesDirectory)/pkg_output/windows/win-x64
```

Comment:
We want those builds to be self-contained. Starting with .NET 8, providing a RID will no longer trigger a self-contained build by default, so we should explicitly pass that in here and below.

Reply:
added `--self-contained`